src/lib.rs

```diff
@@ -176,6 +184,8 @@ impl fmt::Display for Error {
             Error::FmtError => f.write_str("formatter error"),
             #[cfg(feature = "alloc")]
             Error::OutOfMemory(..) => f.write_str("allocation failure"),
+            #[cfg(feature = "std")]
+            Error::IoError(err) => write!(f, "io error: {err:?}"),
```

Comment:
I think it is not idiomatic to format the inner error as part of `impl fmt::Display for ErrorType`.

Per the [`std` docs for `Error`](https://doc.rust-lang.org/stable/std/error/trait.Error.html):

> Errors must describe themselves through the [Display](https://doc.rust-lang.org/stable/std/fmt/trait.Display.html) and [Debug](https://doc.rust-lang.org/stable/std/fmt/trait.Debug.html) traits. Error messages are typically concise lowercase sentences without trailing punctuation:
>
> ```rust
> let err = "NaN".parse::<u32>().unwrap_err();
> assert_eq!(err.to_string(), "invalid digit found in string");
> ```

It looks like none of the error messages here are _sentences_. We may want to fix that separately.